tests/test_execution.py

```diff
@@ -132,13 +132,13 @@ def test_traintuple_execution_failure(factory, client, default_dataset_1):
         dataset=default_dataset_1,
         data_samples=default_dataset_1.train_data_sample_keys,
     )
-    if client.backend == "remote":
+    if client.debug:
```

Comment:
I think we should not have in the test different behaviour for debug and not debug mode.

I would prefer to flag these tests as remote only. 

As this change hasn't been introduced in this PR but we could handle that comment in a separate PR later.

(same comment for next tests)

Reply:
Yes, I think we could modify or move the 'fully local' tests entirely, it is better to do it in another PR as this won't be a single small change.